Xamarin/MvvmSeed/MvvmSeed/MvvmSeed.Android/Setup.cs

```diff
@@ -19,19 +27,29 @@ protected override IMvxIoCProvider CreateIocProvider()
         {
             var containerBuilder = new ContainerBuilder();
             containerBuilder.RegisterModule<ApplicationModule>();
+            containerBuilder.Register(c => Environment.GetFolderPath(Environment.SpecialFolder.Personal)).Named<string>(App.BootstrapParamaters.LocalAppDataFolder);
+            return new AutofacMvxIocProvider(containerBuilder.Build());
+        }
 
-            // This is an important step that ensures all the ViewModel's are loaded into the container. Without it, MvvmCross wouldn't register them by itself; needs more investigation.
-            var viewModelsAssembly = typeof(App).Assembly;
-            containerBuilder.RegisterAssemblyTypes(viewModelsAssembly)
-                .AssignableTo<MvxViewModel>()
-                .As<IMvxViewModel, MvxViewModel>()
-                .AsSelf();
-
-            // TODO [TechnicalDebt | WIP] Improve DbContext initialization
-            // Set platform-specific path to local app data folder
-            containerBuilder.Register(c => System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal)).Named<string>(App.BootstrapParamaters.LocalAppDataFolder);
+        protected override void InitializeViewLookup()
+        {
+            // ViewModel->View relations
+            // Used by MvvmCross framework to automatically resolve ShowViewModel<TViewModel> into a view
+            // MvvmCross by default uses naming convention, but this approach provides flexibility (we can still rely on naming conventions if we want to and override only when required)
+            var viewModelToViewMapping = new Dictionary<Type, Type>
```

Comment:
Yep, if it ends up growing, we should definitely move it to a separate file/method. But I prefer to see how this evolves first, I'm not 100% we really want to do this, rely on naming conventions or use some custom attribute.